ClassesHW56/Accounts/SavingAccount.cs

```diff
@@ -0,0 +1,41 @@
+namespace HW456.ClassesHw56.Accounts;
+
+class SavingAccount : BankAccount
+{
+    public decimal InitialDeposit { get; set; }
+    public decimal Rate { get; set; }
+
+    public override bool DecreaseBalance(decimal amount)
+    {
+        DecreaseBalance(amount);
+        
+        if (amount >= InitialDeposit)
+        {
+            base.DecreaseBalance(amount);
+            return true;
+        }
+        else
+        {
+            throw new ArgumentOutOfRangeException("amount","The amount is bigger than initial deposit");
+        }

```

Comment:
Сначала проверка, затем логика. Так уменьшается вложенность кода, да и чище смотрится:
```suggestion
        if (amount < InitialDeposit)
        {
            throw new ArgumentOutOfRangeException("amount", "The amount is bigger than initial deposit");
        }

        base.DecreaseBalance(amount);

        return true;
```

И `true` здесь возвращать несправедливо по отношению к `base.DecreaseBalance(amount);`, ведь он может и `false` вернуть. В общем, надо возвращать его:
```c#
return base.DecreaseBalance(amount);
```

Reply:
поправила